src/backend/reach_backend/Api/trial_filterer.py

```diff
@@ -27,9 +27,53 @@ def filter_trials(studies: pd.DataFrame, input_params: dict) -> pd.DataFrame:
         ]
 
         studies = TrialFilterer.filter_gender(input_params["sex"], studies)
-        studies["KeywordRank"] = 0
-        studies = TrialFilterer.filter_keywords(studies, input_params)
         return studies
+    
+    @staticmethod
+    def generate_keywords(input_params: dict):
+        search_str = []
+        for k in filtering_dict_num.keys():
+            if k in input_params.keys():
+                if input_params.get(k, 0) > 0:
+                    search_str.append(filtering_dict_num.get(k, 0))
```

Comment:
Both checks here look like they are for the same thing, same in the filter dict loop. If k is in keys doesn't it guarantee that params.get will return a value

Reply:
outer check checks for if the given filter is included in the input parameters, the inner check looks for if it has a value greater than 0 (or True in the case of the boolean filter). While we probably wouldn't include a parameter if it's value is zero/False (if number of exacerbations is 0 why bother including it in the dict?), but it's conceivable that it might be returned by the frontend. The inner check is moreso for verifying that the value isn't zero/False, rather than for checking if the parameter exists at all

Reply:
Makes sense, I was just confused because I assumed the 0 would only show up on the not found case. Looks good to merge then.